packages/common/base/src/apiClient/CrossmintApiClient.ts

```diff
@@ -1,36 +1,45 @@
-import { validateAPIKey } from "@/apiKey/validateAPIKey";
-import { ValidateAPIKeyPrefixExpectations, ValidateAPIKeyPrefixSuccessResult } from "@/apiKey/validateAPIKeyPrefix";
+import { Crossmint } from "@/Crossmint/Crossmint";
+import { environmentToCrossmintBaseURL } from "@/apiKey/utils/environmentToCrossmintBaseURL";
+import { ValidateApiKeySuccessResult, validateAPIKey } from "@/apiKey/validateAPIKey";
+import { ValidateAPIKeyPrefixExpectations } from "@/apiKey/validateAPIKeyPrefix";
 
-import { environmentToCrossmintBaseURL } from "../apiKey/utils/environmentToCrossmintBaseURL";
 import { ApiClient } from "./ApiClient";
 
-export type CrossmintApiClientCtorParams = CrossmintApiClientCtorWithoutExpectationsParams & {
-    expectations?: ValidateAPIKeyPrefixExpectations;
-};
-
-export type CrossmintApiClientCtorWithoutExpectationsParams = {
-    apiKey: string;
-    overrideBaseUrl?: string;
-    extraAuthHeaders?: HeadersInit;
+export type CrossmintApiClientInternalConfig = {
+    sdkMetadata: {
+        name: string;
+        version: string;
+    };
+    apiKeyExpectations?: ValidateAPIKeyPrefixExpectations;
 };
 
 export class CrossmintApiClient extends ApiClient {
-    protected parsedAPIKey: ValidateAPIKeyPrefixSuccessResult;
-    protected overrideBaseUrl?: string;
+    protected parsedAPIKey: ValidateApiKeySuccessResult;
+    protected internalConfig: CrossmintApiClientInternalConfig;
 
-    constructor({ apiKey, expectations, overrideBaseUrl, extraAuthHeaders }: CrossmintApiClientCtorParams) {
-        const apiKeyValidationResult = validateAPIKey(apiKey, expectations);
+    constructor(
+        public crossmint: Crossmint,
+        {
+            internalConfig,
+        }: {
+            internalConfig: CrossmintApiClientInternalConfig;
+        }
+    ) {
+        const apiKeyValidationResult = validateAPIKey(crossmint.config.apiKey, internalConfig.apiKeyExpectations);
```

Comment:
I feel like we should prob do some form of validation when they call new Crossmint({ apiKey }) if its malformed